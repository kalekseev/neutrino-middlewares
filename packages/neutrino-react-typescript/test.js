// eslint-disable-next-line import/no-extraneous-dependencies
import { validate } from 'webpack';
import { Neutrino } from 'neutrino';

// eslint-disable-next-line global-require, import/no-unresolved
const mw = () => require('.');

test('loads preset', () => {
  mw();
});

test('uses preset', () => {
  Neutrino().use(mw());
});

test('valid preset production', () => {
  const api = Neutrino({ env: { NODE_ENV: 'production' } });
  api.use(mw());
  const config = api.config.toConfig();
  const errors = validate(config);
  expect(errors.length).toBe(0);
  expect(config).toMatchSnapshot();
});

test('valid preset development', () => {
  const api = Neutrino({ 'env': { NODE_ENV: 'development' } });
  api.use(mw());
  const config = api.config.toConfig();
  const errors = validate(config);
  expect(errors.length).toBe(0);
  expect(config).toMatchSnapshot();
});
