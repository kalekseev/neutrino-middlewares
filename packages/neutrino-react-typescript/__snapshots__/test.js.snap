// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`valid preset development 1`] = `
Object {
  "context": "/projects/neutrino-middlewares",
  "devServer": Object {
    "contentBase": "/projects/neutrino-middlewares/src",
    "headers": Object {
      "host": "localhost:5000",
    },
    "historyApiFallback": true,
    "host": "0.0.0.0",
    "hot": true,
    "https": false,
    "open": false,
    "overlay": true,
    "port": 5000,
    "public": "localhost:5000",
    "publicPath": "/",
    "stats": Object {
      "assets": false,
      "children": false,
      "chunks": false,
      "colors": true,
      "errorDetails": true,
      "errors": true,
      "hash": false,
      "modules": false,
      "publicPath": false,
      "timings": false,
      "version": false,
      "warnings": true,
    },
  },
  "devtool": "cheap-module-source-map",
  "entry": Object {
    "index": Array [
      "/projects/neutrino-middlewares/node_modules/webpack-dev-server/client/index.js?http://localhost:5000",
      "/projects/neutrino-middlewares/node_modules/webpack/hot/dev-server.js",
      "/projects/neutrino-middlewares/node_modules/react-hot-loader/patch.js",
      "/projects/neutrino-middlewares/src/index",
    ],
  },
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.\\(html\\)\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/html-loader/index.js",
            "options": Object {
              "name": "[name].[ext]",
            },
          },
        ],
      },
      Object {
        "exclude": Array [
          /\\\\\\.module\\.s\\?css\\$/,
        ],
        "test": /\\\\\\.s\\?css\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/style-loader/index.js",
            "options": Object {
              "hmr": true,
            },
          },
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/css-loader/index.js",
            "options": Object {
              "importLoaders": 0,
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.module\\.s\\?css\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/style-loader/index.js",
            "options": Object {
              "hmr": true,
            },
          },
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/css-loader/index.js",
            "options": Object {
              "importLoaders": 0,
              "modules": true,
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.\\(woff\\|woff2\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/url-loader/index.js",
            "options": Object {
              "limit": 10000,
              "mimetype": "application/font-woff",
              "name": "[name].[ext]",
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.ttf\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/url-loader/index.js",
            "options": Object {
              "limit": 10000,
              "mimetype": "application/octet-stream",
              "name": "[name].[ext]",
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.eot\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/file-loader/dist/cjs.js",
            "options": Object {
              "name": "[name].[ext]",
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.svg\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/url-loader/index.js",
            "options": Object {
              "limit": 8192,
              "name": "[name].[ext]",
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.\\(png\\|jpg\\|jpeg\\|gif\\|webp\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/url-loader/index.js",
            "options": Object {
              "limit": 8192,
              "name": "[name].[ext]",
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.ico\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/url-loader/index.js",
            "options": Object {
              "limit": 8192,
              "name": "[name].[ext]",
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.\\(worker\\.js\\|worker\\.jsx\\|worker\\.vue\\|worker\\.ts\\|worker\\.tsx\\|worker\\.mjs\\)\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/worker-loader/dist/cjs.js",
          },
        ],
      },
      Object {
        "test": /\\\\\\.tsx\\?\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/awesome-typescript-loader/dist/entry.js",
          },
        ],
      },
      Object {
        "enforce": "pre",
        "test": /\\\\\\.tsx\\?\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/tslint-loader/index.js",
          },
        ],
      },
      Object {
        "enforce": "pre",
        "test": /\\\\\\.js\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/source-map-loader/index.js",
          },
        ],
      },
    ],
  },
  "node": Object {
    "Buffer": false,
    "fs": "empty",
    "tls": "empty",
  },
  "output": Object {
    "chunkFilename": "[name].[chunkhash].js",
    "devtoolModuleFilenameTemplate": [Function],
    "filename": "[name].js",
    "pathinfo": true,
    "publicPath": "/",
  },
  "plugins": Array [
    EnvironmentPlugin {
      "defaultValues": Object {},
      "keys": Array [
        "NODE_ENV",
      ],
    },
    HtmlWebpackPlugin {
      "options": Object {
        "appMountId": "root",
        "cache": true,
        "chunks": Array [
          "index",
          "vendor",
          "runtime",
        ],
        "compile": true,
        "excludeChunks": Array [],
        "favicon": false,
        "filename": "index.html",
        "hash": false,
        "inject": false,
        "minify": Object {
          "collapseWhitespace": true,
          "keepClosingSlash": true,
          "preserveLineBreaks": true,
          "useShortDoctype": true,
        },
        "mobile": true,
        "pluginId": "html-index",
        "showErrors": true,
        "template": "/projects/neutrino-middlewares/node_modules/html-webpack-template/index.ejs",
        "title": "Webpack App",
        "xhtml": true,
      },
    },
    HotModuleReplacementPlugin {
      "fullBuildTimeout": 200,
      "multiStep": undefined,
      "options": Object {},
      "requestTimeout": 10000,
    },
    NamedModulesPlugin {
      "options": Object {},
    },
    CheckerPlugin {},
  ],
  "resolve": Object {
    "alias": Object {
      "react-native": "react-native-web",
    },
    "extensions": Array [
      ".js",
      ".jsx",
      ".vue",
      ".ts",
      ".tsx",
      ".mjs",
      ".json",
    ],
    "modules": Array [
      "node_modules",
      "/projects/neutrino-middlewares/node_modules",
      "/projects/neutrino-middlewares/packages/neutrino-web/node_modules",
      "/projects/neutrino-middlewares/packages/neutrino-react-typescript/node_modules",
    ],
  },
  "resolveLoader": Object {
    "modules": Array [
      "/projects/neutrino-middlewares/node_modules",
      "/projects/neutrino-middlewares/packages/neutrino-web/node_modules",
      "/projects/neutrino-middlewares/packages/neutrino-react-typescript/node_modules",
    ],
  },
  "target": "web",
}
`;

exports[`valid preset production 1`] = `
Object {
  "bail": true,
  "context": "/projects/neutrino-middlewares",
  "devtool": "source-map",
  "entry": Object {
    "index": Array [
      "/projects/neutrino-middlewares/src/index",
    ],
  },
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.\\(html\\)\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/html-loader/index.js",
            "options": Object {
              "name": "[name].[ext]",
            },
          },
        ],
      },
      Object {
        "exclude": Array [
          /\\\\\\.module\\.s\\?css\\$/,
        ],
        "test": /\\\\\\.s\\?css\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/css-hot-loader/index.js",
          },
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/extract-text-webpack-plugin/dist/loader.js",
            "options": Object {
              "omit": 1,
              "remove": true,
            },
          },
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/style-loader/index.js",
            "options": Object {
              "hmr": false,
            },
          },
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/css-loader/index.js",
            "options": Object {
              "importLoaders": 0,
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.module\\.s\\?css\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/css-hot-loader/index.js",
          },
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/extract-text-webpack-plugin/dist/loader.js",
            "options": Object {
              "omit": 1,
              "remove": true,
            },
          },
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/style-loader/index.js",
            "options": Object {
              "hmr": false,
            },
          },
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/css-loader/index.js",
            "options": Object {
              "importLoaders": 0,
              "modules": true,
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.\\(woff\\|woff2\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/url-loader/index.js",
            "options": Object {
              "limit": 10000,
              "mimetype": "application/font-woff",
              "name": "[name].[hash].[ext]",
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.ttf\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/url-loader/index.js",
            "options": Object {
              "limit": 10000,
              "mimetype": "application/octet-stream",
              "name": "[name].[hash].[ext]",
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.eot\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/file-loader/dist/cjs.js",
            "options": Object {
              "name": "[name].[hash].[ext]",
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.svg\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/url-loader/index.js",
            "options": Object {
              "limit": 8192,
              "name": "[name].[hash].[ext]",
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.\\(png\\|jpg\\|jpeg\\|gif\\|webp\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/url-loader/index.js",
            "options": Object {
              "limit": 8192,
              "name": "[name].[hash].[ext]",
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.ico\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/url-loader/index.js",
            "options": Object {
              "limit": 8192,
              "name": "[name].[hash].[ext]",
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.\\(worker\\.js\\|worker\\.jsx\\|worker\\.vue\\|worker\\.ts\\|worker\\.tsx\\|worker\\.mjs\\)\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/worker-loader/dist/cjs.js",
          },
        ],
      },
      Object {
        "test": /\\\\\\.tsx\\?\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/awesome-typescript-loader/dist/entry.js",
          },
        ],
      },
      Object {
        "enforce": "pre",
        "test": /\\\\\\.tsx\\?\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/tslint-loader/index.js",
          },
        ],
      },
      Object {
        "enforce": "pre",
        "test": /\\\\\\.js\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/source-map-loader/index.js",
          },
        ],
      },
    ],
  },
  "node": Object {
    "Buffer": false,
    "fs": "empty",
    "tls": "empty",
  },
  "output": Object {
    "chunkFilename": "[name].[chunkhash].js",
    "devtoolModuleFilenameTemplate": [Function],
    "filename": "[name].[chunkhash:8].js",
    "path": "/projects/neutrino-middlewares/build",
    "pathinfo": true,
    "publicPath": "/",
  },
  "plugins": Array [
    EnvironmentPlugin {
      "defaultValues": Object {},
      "keys": Array [
        "NODE_ENV",
      ],
    },
    HtmlWebpackPlugin {
      "options": Object {
        "appMountId": "root",
        "cache": true,
        "chunks": Array [
          "index",
          "vendor",
          "runtime",
        ],
        "compile": true,
        "excludeChunks": Array [],
        "favicon": false,
        "filename": "index.html",
        "hash": false,
        "inject": false,
        "minify": Object {
          "collapseWhitespace": true,
          "keepClosingSlash": true,
          "preserveLineBreaks": true,
          "useShortDoctype": true,
        },
        "mobile": true,
        "pluginId": "html-index",
        "showErrors": true,
        "template": "/projects/neutrino-middlewares/node_modules/html-webpack-template/index.ejs",
        "title": "Webpack App",
        "xhtml": true,
      },
    },
    ExtractTextPlugin {
      "filename": "[name].[contenthash:8].css",
      "id": 1,
      "options": Object {
        "allChunks": true,
        "ignoreOrder": true,
      },
    },
    ExtractTextPlugin {
      "filename": "[name].[contenthash:8].css",
      "id": 2,
      "options": Object {
        "allChunks": true,
        "ignoreOrder": true,
      },
    },
    NamedModulesPlugin {
      "options": Object {},
    },
    NamedChunksPlugin {
      "nameResolver": [Function],
    },
    CommonsChunkPlugin {
      "async": undefined,
      "children": undefined,
      "chunkNames": Array [
        "vendor",
      ],
      "deepChildren": undefined,
      "filenameTemplate": undefined,
      "ident": "/projects/neutrino-middlewares/node_modules/webpack/lib/optimize/CommonsChunkPlugin.js0",
      "minChunks": Infinity,
      "minSize": undefined,
      "selectedChunks": undefined,
    },
    CommonsChunkPlugin {
      "async": undefined,
      "children": undefined,
      "chunkNames": Array [
        "runtime",
      ],
      "deepChildren": undefined,
      "filenameTemplate": undefined,
      "ident": "/projects/neutrino-middlewares/node_modules/webpack/lib/optimize/CommonsChunkPlugin.js1",
      "minChunks": undefined,
      "minSize": undefined,
      "selectedChunks": undefined,
    },
    NameAllModulesPlugin {},
    ModuleConcatenationPlugin {
      "options": Object {},
    },
    CleanWebpackPlugin {
      "options": Object {
        "allowExternal": false,
        "dry": false,
        "root": "/projects/neutrino-middlewares",
        "verbose": false,
      },
      "paths": Array [
        "/projects/neutrino-middlewares/build",
      ],
    },
    Object {
      "apply": [Function],
    },
  ],
  "resolve": Object {
    "alias": Object {
      "react-native": "react-native-web",
    },
    "extensions": Array [
      ".js",
      ".jsx",
      ".vue",
      ".ts",
      ".tsx",
      ".mjs",
      ".json",
    ],
    "modules": Array [
      "node_modules",
      "/projects/neutrino-middlewares/node_modules",
      "/projects/neutrino-middlewares/packages/neutrino-web/node_modules",
      "/projects/neutrino-middlewares/packages/neutrino-react-typescript/node_modules",
    ],
  },
  "resolveLoader": Object {
    "modules": Array [
      "/projects/neutrino-middlewares/node_modules",
      "/projects/neutrino-middlewares/packages/neutrino-web/node_modules",
      "/projects/neutrino-middlewares/packages/neutrino-react-typescript/node_modules",
    ],
  },
  "target": "web",
}
`;
